pkg/apis/duck/v1alpha1/subscribable_types.go

```diff
@@ -34,6 +35,9 @@ type Subscribable struct {
 	Subscribers []ChannelSubscriberSpec `json:"subscribers,omitempty" patchStrategy:"merge" patchMergeKey:"uid"`
 }
 
+// Subscribable is an Implementable "duck type".
```

Comment:
I think we should create a new Duck Type that looks something like this:

type Channelable struct {
  metav1.TypeMeta ...
  metav1.ObjectMeta ...
 Spec ChannelSpec ...
 Status Addressable ...
 
}

So that we have one Duck type that allows us to create Subscriptions to it as well get the Addressable part, since a Channel is _both_ so rather than muck with addressable and subscribable, we have a proper Channelable type that handles spec / status. I think Addressable might not be enough, since we might also need some kind of Subscriptions statuses wired through. If you think this is reasonable, let's create an issue to track this, unless we think this change might be useful to do now.
@Harwayne @n3wscott

Reply:
I agree we need this to make things cleaner. 
IMO we can track it and do it in a follow up, but waiting to hear thoughts from the guys...

Reply:
fine with an issue, we need to relax this constraint to be able to use new channels, please generate an issue to track this work and I'm good.

Reply:
https://github.com/knative/eventing/issues/1292